google/cloud/bigtable/mutations_batcher.py

```diff
@@ -46,61 +172,246 @@ class MutationsBatcher:
+        self.closed: bool = False
+        self._table = table
+        self._staged_entries: list[RowMutationEntry] = []
+        self._staged_count, self._staged_bytes = 0, 0
+        self._flow_control = _FlowControl(
+            flow_control_max_count, flow_control_max_bytes
+        )
+        self._flush_limit_bytes = flush_limit_bytes
+        self._flush_limit_count = (
+            flush_limit_mutation_count
+            if flush_limit_mutation_count is not None
+            else float("inf")
+        )
+        self.exceptions: list[Exception] = []
+        self._flush_timer_task: asyncio.Task[None] = asyncio.create_task(
+            self._flush_timer(flush_interval)
+        )
+        # create noop previous flush task to avoid None checks
+        self._prev_flush: asyncio.Task[None] = asyncio.create_task(asyncio.sleep(0))
+        # MutationExceptionGroup reports number of successful entries along with failures
+        self._entries_processed_since_last_raise: int = 0
+        # keep track of entries that are set to be sent in next flush,
+        # so we can add more before if mutations are added before flush task starts
+        self._scheduled_flush_entries: list[RowMutationEntry] = []
 
-    async def append(self, row_key: str | bytes, mutation: Mutation | list[Mutation]):
+    async def _flush_timer(self, interval: float | None):
         """
-        Add a new mutation to the internal queue
+        Triggers new flush tasks every `interval` seconds
         """
-        raise NotImplementedError
+        if interval is None:
+            return
+        while not self.closed:
+            await asyncio.sleep(interval)
+            # add new flush task to list
+            if not self.closed and self._staged_entries:
+                self._schedule_flush()
 
-    async def append_conditional(
-        self,
-        predicate_filter: RowFilter,
-        row_key: str | bytes,
-        if_true_mutations: Mutation | list[Mutation] | None = None,
-        if_false_mutations: Mutation | list[Mutation] | None = None,
-    ):
+    def append(self, mutation_entry: RowMutationEntry):
+        """
+        Add a new set of mutations to the internal queue
+
+        Args:
+          - mutation_entry: new entry to add to flush queue
+        Raises:
+          - RuntimeError if batcher is closed
+          - ValueError if an invalid mutation type is added
+        """
+        if self.closed:
+            raise RuntimeError("Cannot append to closed MutationsBatcher")
+        if isinstance(mutation_entry, Mutation):  # type: ignore
+            raise ValueError(
+                f"invalid mutation type: {type(mutation_entry).__name__}. Only RowMutationEntry objects are supported by batcher"
+            )
+        if self._scheduled_flush_entries:
+            # flush is already scheduled to run on next loop iteration
+            # add new entries directly to flush list
+            self._scheduled_flush_entries.append(mutation_entry)
```

Comment:
Although I guess another way to solve the problem would be to make `append` into an async coroutine, and yield immediately to the event loop in there when the flush limits are hit. There are pros and cons, but it would make the code a bit simpler. Let me know what you think

Reply:
I ended up making the `async append` change, since it should simplify things for the sync surface.